client/allocwatcher/alloc_watcher_unix_test.go

```diff
@@ -136,6 +136,60 @@ func TestPrevAlloc_StreamAllocDir_BadSymlink_Linkname(t *testing.T) {
 	must.EqError(t, err, "archive contains symlink that escapes alloc dir")
 }
 
+func TestPrevAlloc_StreamAllocDir_SyminkWriteAttack(t *testing.T) {
+	ci.Parallel(t)
+
+	tmpDir := t.TempDir()
+	outsidePath := filepath.Join(tmpDir, "outside")
+	insidePath := "malformed_link"
+	content := "HelloWorld from outside"
+
+	// Create a tar archive with a symlink that attempts to escape the allocation directory
```

Comment:
For future developers, it's important to understand that this isn't just as matter of a symlink which escapes the allocation directory, but one where the second header will write to the same path and follow the symlink out of the sandbox. We should expand upon that in this comment.

Reply:
Good idea, added more details here https://github.com/hashicorp/nomad/pull/23738/commits/4b169330e97c48556713493c7ad644c59b4f30a1